### PR TITLE
BFD-2766: Fix unit tests to use mocks of drug code and npi org lookup

### PR DESCRIPTION
**Pull request by rdatoc**
**JIRA Ticket:**
[BFD-2766](https://jira.cms.gov/browse/BFD-2766)

**User Story or Bug Summary:**

Updated unit tests to use mocks of drug code and npi org lookup

---

### What Does This PR Do?

Mock lookup classes for unit tests:

- Create static mocks for FdaDrugCodeDisplayLookup and NPIOrgLookup
- Mock the static setup method to return mocks of the data
- Additionally, replace instantiations of NPIOrgLookup with a static method to return test data

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by dondevun — COMMENTED** [review]
I'm not crazy about the pattern that exists of having real classes returning test data, but I think it will be ok.

**Review by dondevun — DISMISSED** [review]

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2Test.java [hunk 1]
```diff
@@ -116,12 +126,32 @@ public CarrierClaim generateClaim() throws FHIRException {
   public void before() throws IOException {
     when(metricRegistry.timer(any())).thenReturn(metricsTimer);
     when(metricsTimer.time()).thenReturn(metricsTimerContext);
+    npiOrgLookup = Mockito.mockStatic(NPIOrgLookup.class);
```
Comment (aschey-forpeople) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2Test.java:129]: This setup logic is copy/pasted a bunch of times. Should we maybe consider adding a helper utility to centralize some of these things? We already have a utils folder [here](https://github.com/CMSgov/beneficiary-fhir-data/tree/master/apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/utils)

File: apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java [hunk 2]
```diff
@@ -83,6 +83,15 @@ public static FdaDrugCodeDisplayLookup createDrugCodeLookupForProduction() {
     return drugCodeLookupForProduction;
   }
 
+  /**
+   * Constructs a {@link FdaDrugCodeDisplayLookup} for testing purposes only.
+   *
+   * @param npiOrgMap Drug code lookup map to populate FdaDrugCodeDisplayLookup
+   */
+  public FdaDrugCodeDisplayLookup(Map<String, String> npiOrgMap) {
```
Comment (JFU-NAVA-PBC) [inline apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java:91]: nit: parameter name could use NDC related ? like ndcMap
same for the java doc @param

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2Test.java [hunk 3]
```diff
@@ -795,7 +832,7 @@ public void shouldHaveCareTeamMembers() {
             "http://terminology.hl7.org/CodeSystem/claimcareteamrole",
             "primary",
             "Primary provider");
-    compare4.getProvider().setDisplay("Fake ORG Name");
+    compare4.getProvider().setDisplay(NPIOrgLookup.FAKE_NPI_ORG_NAME);
```
Comment (JFU-NAVA-PBC) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2Test.java:835]: good style to not to use literal

**Review by JFU-NAVA-PBC — APPROVED** [review]
LGTM, 

Like the code improvements here and there.

Noticed BFD CI does not do Sonar coverage, otherwise, would check coverage impact of the re-factor, should stay the same.

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2Test.java (hunk 3 above)
Comment (rdatoc) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2Test.java:816]: thanks

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2Test.java (hunk 1 above)
Comment (rdatoc) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2Test.java]: done

File: apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java (hunk 2 above)
Comment (rdatoc) [reply apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java:91]: too much copy pasta.  i'll rename

Comment (rdatoc) [reply apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java]: done

**Review by JFU-NAVA-PBC — APPROVED** [review]
LGTM

Tested on local setup

AC criteria met plus coding style improvements

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/PartDEventTransformerV2Test.java [hunk 4]
```diff
@@ -105,6 +112,17 @@ public PartDEvent generateClaim() throws FHIRException {
   public void before() throws IOException {
     when(metricRegistry.timer(any())).thenReturn(metricsTimer);
     when(metricsTimer.time()).thenReturn(metricsTimerContext);
+    fdaDrugCodeDisplayLookup = Mockito.mockStatic(FdaDrugCodeDisplayLookup.class);
```
Comment (aschey-forpeople) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/PartDEventTransformerV2Test.java:115]: Also here

**Review by dondevun — APPROVED** [review]

**Pull request merged**